src/routes/github/github-oauth-router.ts

```diff
@@ -38,8 +39,8 @@ const GithubOAuthLoginGet = async (req: Request, res: Response): Promise<void> =
 
 	// Save the redirect that may have been specified earlier into session to be retrieved later
 	req.session[state] =
-		res.locals.redirect ||
-		`/github/configuration${url.parse(req.originalUrl).search || ""}`;
+		appendJiraHostIfNeeded(res.locals.redirect ||
+		`/github/configuration${url.parse(req.originalUrl).search || ""}`, res.locals.jiraHost);
```

Comment:
Instead of appending `jiraHost`, I think it would be much cleaner if we get `jirahost` from query parameter or session in `TempGetJiraHostFromStateMiddleware`

Reply:
will jirahost exist in locals at this point since its a new request?

@Harminder84 - Session would work but I think we were trying to avoid keeping a unsafe jirahost in the session. 

We may want to add a conditional that we only adding this as a param for the damn create-branch route.


Alternatively we could put it in the session then rip it out again before we redirect to create-branch route, but sounds messy just typing it.

Reply:
So this state is stored in session, and for create-branch, it get the jiraHost from query (unsafe), and for normal flow, it get it from jiraHostMiddleware (safe)

The line here is to save the `state` ( redirect url that contains jiraHost ) in session before OAuth callback. The state is empty here, so we can't get it from state.

Reply:
Also state get removed after callback. So only unsafe part is Jira sending jiraHost to our app in query string during create-branch. That's our agreement for unsafe part right?